lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -2032,7 +2045,8 @@ export class UserAgentApplication {
         // acquireTokenSilent
         if (silentCall) {
             if (Account.compareAccounts(accountObject, this.getAccount())) {
-                tokenType = (scopes.indexOf(this.config.auth.clientId) > -1) ? ResponseTypes.id_token : ResponseTypes.token;
+                const idTokenRequested = scopes.indexOf(this.config.auth.clientId) > -1 || scopes.indexOf("openid") > -1; // maybe add profile
```

Comment:
If this is done after scopes are validated/set we don't need clientId here and can just stick to openid/profile

Reply:
Right, the next PR will be dedicated to updating the getTokenType logic to include all the response type cases we've discussed.